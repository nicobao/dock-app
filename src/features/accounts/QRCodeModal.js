import {Box, Stack} from 'native-base';
import React from 'react';
import {Dimensions} from 'react-native';
import QRCode from 'react-native-qrcode-svg';
import {Modal} from '../../components/Modal';
import {DockLogoIcon, Typography, Theme, Button} from '../../design-system';
import {translate} from '../../locales';

export function QRCodeModal({onClose, visible, data, title, description}) {
  const windowWidth = Dimensions.get('window').width;

  return (
    <Modal visible={visible} onClose={onClose} modalSize={0.95}>
      <Stack p={8}>
        <Typography variant="h1">{title}</Typography>
        <Typography>{description}</Typography>
        <Box
          mt={8}
          alignItems="center"
          p={3}
          bg={Theme.colors.qrCodeBackground}>
          {data && (
            <QRCode value={data} logo={DockLogoIcon} size={windowWidth * 0.8} />
          )}
        </Box>
        <Stack mt={5}>
          <Button variant="solid" size="sm" onPress={onClose}>
            {translate('navigation.close')}
          </Button>
        </Stack>
      </Stack>
    </Modal>
  );
}
