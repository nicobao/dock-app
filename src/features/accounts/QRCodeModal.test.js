import React from 'react';
import {shallow} from 'enzyme';
import {QRCodeModal} from './QRCodeModal';

describe('QRCodeModal', () => {
  it('should render correctly with data', () => {
    const wrapper = shallow(
      <QRCodeModal onClose={jest.fn()} data="some data to be rendered" />,
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should not break when data is undefined', () => {
    const wrapper = shallow(
      <QRCodeModal onClose={jest.fn()} data={undefined} />,
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
