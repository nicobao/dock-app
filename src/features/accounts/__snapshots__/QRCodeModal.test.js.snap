// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRCodeModal should not break when data is undefined 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  coverScreen={true}
  customBackdrop={null}
  deviceHeight={null}
  deviceWidth={null}
  hasBackdrop={true}
  hideModalContentWhileAnimating={false}
  isVisible={false}
  onBackButtonPress={[MockFunction]}
  onBackdropPress={[MockFunction]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onSwipeComplete={[MockFunction]}
  panResponderThreshold={4}
  propagateSwipe={false}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  style={
    Object {
      "justifyContent": "flex-end",
      "margin": 0,
    }
  }
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection={
    Array [
      "down",
    ]
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "backgroundColor": "#18181B",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 1267.3,
        "width": "100%",
      }
    }
  >
    <Memo(ForwardRef(Stack))
      alignItems="center"
      mt={6}
    >
      <Memo(ForwardRef(Box))
        backgroundColor="#3F3F46"
        borderRadius={6}
        height="4px"
        width="36px"
      />
    </Memo(ForwardRef(Stack))>
    <Memo(ForwardRef(Stack))
      p={8}
    >
      <Styled(Component)
        variant="h1"
      />
      <Styled(Component) />
      <Memo(ForwardRef(Box))
        alignItems="center"
        bg="#fff"
        mt={8}
        p={3}
      />
      <Memo(ForwardRef(Stack))
        mt={5}
      >
        <Button
          onPress={[MockFunction]}
          size="sm"
          variant="solid"
        >
          Close
        </Button>
      </Memo(ForwardRef(Stack))>
    </Memo(ForwardRef(Stack))>
  </View>
</ReactNativeModal>
`;

exports[`QRCodeModal should render correctly with data 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  coverScreen={true}
  customBackdrop={null}
  deviceHeight={null}
  deviceWidth={null}
  hasBackdrop={true}
  hideModalContentWhileAnimating={false}
  isVisible={false}
  onBackButtonPress={[MockFunction]}
  onBackdropPress={[MockFunction]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onSwipeComplete={[MockFunction]}
  panResponderThreshold={4}
  propagateSwipe={false}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  style={
    Object {
      "justifyContent": "flex-end",
      "margin": 0,
    }
  }
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection={
    Array [
      "down",
    ]
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "backgroundColor": "#18181B",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 1267.3,
        "width": "100%",
      }
    }
  >
    <Memo(ForwardRef(Stack))
      alignItems="center"
      mt={6}
    >
      <Memo(ForwardRef(Box))
        backgroundColor="#3F3F46"
        borderRadius={6}
        height="4px"
        width="36px"
      />
    </Memo(ForwardRef(Stack))>
    <Memo(ForwardRef(Stack))
      p={8}
    >
      <Styled(Component)
        variant="h1"
      />
      <Styled(Component) />
      <Memo(ForwardRef(Box))
        alignItems="center"
        bg="#fff"
        mt={8}
        p={3}
      >
        <QRCode
          logo={[Function]}
          size={600}
          value="some data to be rendered"
        />
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Stack))
        mt={5}
      >
        <Button
          onPress={[MockFunction]}
          size="sm"
          variant="solid"
        >
          Close
        </Button>
      </Memo(ForwardRef(Stack))>
    </Memo(ForwardRef(Stack))>
  </View>
</ReactNativeModal>
`;
